src/App.tsx

```diff
@@ -1,61 +1,139 @@
+  const userDefaultValue = 'Choose a user';
+
+  const [title, setTitle] = useState(titleDefaultValue);
+  const [user, setUser] = useState(userDefaultValue);
+  const [todos, setTodos] = useState(todoList);
+  const [isUserSelected, setIsUserSelected] = useState(true);
+  const [isTitleEntered, setIsTitleEntered] = useState(true);
+
+  const handleSubmitForm = (event: React.FormEvent) => {
+    event.preventDefault();
+    if (user !== userDefaultValue && title !== titleDefaultValue) {
+      const newUser = getUserByName(user, usersFromServer);
+      const newTodoId = getLargestTodoId(todos) + 1;
+      const newTodo: Todo = {
+        id: newTodoId,
+        title,
+        completed: false,
+        userId: newUser ? newUser.id : null,
+        user: newUser,
+      };
+
+      setTodos(currentTodo => [...currentTodo, newTodo]);
+      setTitle(titleDefaultValue);
+      setUser(userDefaultValue);
+    }
+
+    if (user === userDefaultValue) {
+      setIsUserSelected(false);
+    }
+
+    if (title === titleDefaultValue) {
+      setIsTitleEntered(false);
+    }
+  };
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.currentTarget.value);
+    setIsTitleEntered(true);
+  };
+
+  const handleUserChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUser(event.currentTarget.value);
+    setIsUserSelected(true);
+  };
 
-export const App = () => {
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form
+        action="/api/users"
+        method="POST"
+        onSubmit={handleSubmitForm}
+      >
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label htmlFor="titleInput">
+            {'Title: '}
+          </label>
+
+          <input
+            type="text"
+            id="titleInput"
+            name="titleInput"
+            data-cy="titleInput"
+            placeholder="Enter a title"
+            value={title}
+            onChange={handleTitleChange}
+            autoComplete="off"
+          />
+          {!isTitleEntered && (
+            <span className="error">Please enter a title</span>
+          )}
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>Choose a user</option>
+          <label htmlFor="userSelect">
+            {'User: '}
+          </label>
+
+          <select
+            id="userSelect"
+            name="userSelect"
+            data-cy="userSelect"
+            value={user}
+            onChange={handleUserChange}
+          >
+            <option value={userDefaultValue} disabled>
+              {userDefaultValue}
+            </option>
+            {usersFromServer.map(userInfo => (
+              <option value={userInfo.name} key={userInfo.id}>
```

Comment:
You could set id as value, but display the user name
```suggestion
              <option value={userInfo.id} key={userInfo.id}>
```

Reply:
I think it is better (more convenient) to use name here, because event.currentTarget.value from onChange event handler always returns string.

Reply:
Yes, it returns the string. But you don't need to create a new function for getting user by name